### PR TITLE
refactor: remove CmdArgParser::ToUpper()

### PR DESCRIPTION
**Pull request by BorysTheDev**
fixes: #3471
the code was refactored to avoid using CmdArgParser::ToUpper()

### CODE REVIEW

**Comment by kostasrim** [comment]
:dancers: :beach_umbrella:

File: src/server/bitops_family.cc [hunk 1]
```diff
@@ -958,25 +958,24 @@ OpResult<std::vector<ResultType>> StateExecutor::Execute(const CommandList& comm
...
+
+  auto [encoding, offset_str] = parser->Next<string_view, string_view>();
+
+  if (encoding.size() < 2) {
     return make_unexpected(kSyntaxErr);
   }
-
-  auto encoding = parser.ToUpper().Next();
-  if (absl::StartsWith(encoding, "U")) {
+  if (encoding[0] == 'U' || encoding[0] == 'u') {
```
Comment (kostasrim) [inline src/server/bitops_family.cc:970]: What's wrong with `absl::EqualsIgnoreCase` ? :eyes:

**Review by kostasrim — APPROVED** [review]

File: src/server/bitops_family.cc (hunk 1 above)
Comment (kostasrim) [reply src/server/bitops_family.cc:970]: string comparison is character by character. if you do `absl::EqualIgnoreCase(enconding[0], 'U')` it should be the exact same comparison under the hood

Comment (BorysTheDev) [reply src/server/bitops_family.cc:970]: EqualsIgnoreCase(absl::string_view piece1, absl::string_view piece2)
I don't see interface for char comparison

Comment (kostasrim) [reply src/server/bitops_family.cc:970]: a char is a `string_view` of size 1 :) Internally it should compare it character by character

**Review by kostasrim — APPROVED** [review]

File: src/server/bitops_family.cc (hunk 1 above)
Comment (BorysTheDev) [reply src/server/bitops_family.cc:970]: it's overkill)

Comment (kostasrim) [reply src/server/bitops_family.cc:970]: `std::string_view tmp = "t";` // tmp has now size 1

Comment (kostasrim) [reply src/server/bitops_family.cc:970]: > it's overkill)

I wouldn't say so but do whatever you like, I already accepted the PR :)

**Pull request merged**